File: Data Structures and Algorithms/4. Recursion/Ankon Sir Problems/Common Problems/_4check_palindrome.cpp
#include <iostream>
using namespace std;

void recursive(string& s, int f, int l) {
    if(f == s.length() / 2) {
        return;
    } else {
        swap(s[f], s[l]);
        recursive(s, f++, l--);
    }
}
void reverse(string& s) {
    int n = s.length();
    for (int i = 0; i < n / 2; i++) {
        swap(s[i], s[n - 1 - i]);
    }
}

int main() {
    string s = "rahim";
    int n = s.length();
    recursive(s, 0, n - 1);
    cout << s << '\n';
    return 0;
}

File: Data Structures and Algorithms/4. Recursion/Ankon Sir Problems/Common Problems/_4check_palindrome.java
import java.util.*;
class _4check_palindrome {
    public static String reverse(String str, int i) {
        if(i == str.length()) return str;
        swap(str[i], str[str.length - 1]);
        reverse(str, i++);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String str;
        str = sc.nextInt();
        if(reverse(str) ) {

        }
    }
}
